File: backend/src/main/java/com/bookkeeping/entity/Deposit.java
package com.bookkeeping.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.GenericGenerator;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "deposits")
@Data
public class Deposit {
    @Id
    @GeneratedValue(generator = "increment")
    @GenericGenerator(name = "increment", type = org.hibernate.id.IncrementGenerator.class)
    private Long id;

    @Column(nullable = false)
    private Long userId;

    @Column(nullable = false)
    private Long accountId;

    @Column(nullable = false, length = 50)
    private String depositType;

    @Column(nullable = false)
    private LocalDate depositTime;

    @Column(nullable = false, precision = 15, scale = 2)
    private BigDecimal amount;

    @Column(precision = 5, scale = 2)
    private BigDecimal interestRate;

    @Column
    private Integer term;

    @Column(columnDefinition = "TEXT")
    private String note;

    @Column(nullable = false)
    private LocalDate reconciliationDate;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: frontend/src/api/deposit.ts
import request from './request'

export interface Deposit {
  id: number
  accountId: number
  depositType: string
  depositTime: string
  amount: number
  interestRate?: number
  term?: number
  note?: string
  reconciliationDate: string
}

export interface CreateDepositRequest {
  accountId: number
  depositType: string
  depositTime: string
  amount: number
  interestRate?: number
  term?: number
  note?: string
  reconciliationDate: string
}

export interface UpdateDepositRequest {
  depositType: string
  depositTime: string
  amount: number
  interestRate?: number
  term?: number
  note?: string
}

export const depositApi = {
  // 获取账户的存款记录
  getDepositsByAccount(accountId: number, date?: string) {
    const params: any = {}
    if (date) {
      params.date = date
    }
    return request.get<Deposit[]>(`/accounts/${accountId}/deposits`, { params })
  },

  // 创建存款记录
  createDeposit(data: CreateDepositRequest) {
    return request.post<Deposit>('/deposits', data)
  },

  // 更新存款记录
  updateDeposit(id: number, data: UpdateDepositRequest) {
    return request.put<Deposit>(`/deposits/${id}`, data)
  },

  // 删除存款记录
  deleteDeposit(id: number) {
    return request.delete(`/deposits/${id}`)
  }
}
